third_party/blink/renderer/core/resource_pool_limiter/resource_pool_limiter.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_THIRD_PARTY_BLINK_RENDERER_CORE_RESOURCE_POOL_LIMITER_RESOURCE_POOL_LIMITER_H_
+#define BRAVE_THIRD_PARTY_BLINK_RENDERER_CORE_RESOURCE_POOL_LIMITER_RESOURCE_POOL_LIMITER_H_
+
+#include <memory>
+#include <utility>
+
+#include "third_party/blink/renderer/core/core_export.h"
+#include "third_party/blink/renderer/platform/wtf/hash_map.h"
+#include "third_party/blink/renderer/platform/wtf/text/string_hash.h"
+#include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
+#include "third_party/blink/renderer/platform/wtf/threading_primitives.h"
+
+namespace blink {
+
+class ExecutionContext;
+
+class CORE_EXPORT ResourcePoolLimiter {
+ public:
+  enum class ResourceType {
+    kWebSocket,
+  };
+
+  class CORE_EXPORT ResourceInUseTracker {
+   public:
+    explicit ResourceInUseTracker(String resource_id_in_use);
+    ~ResourceInUseTracker();
+
+    const String& resource_id_in_use() const { return resource_id_in_use_; }
+
+   private:
+    String resource_id_in_use_;
```

Comment:
Nit: seems like this and the getter could just be called resource_id_/resource_id()?